Src/PerformanceCollector/Shared/QuickPulseTelemetryProcessor.cs

```diff
@@ -157,13 +165,78 @@ public void Process(ITelemetry telemetry)
             }
             finally
             {
-                if (letItemThrough)
+                if (letTelemetryThrough)
                 {
                     this.Next.Process(telemetry);
                 }
             }
         }
 
+        private static ITelemetryDocument ConvertTelemetryToTelemetryDocument(
+            RequestTelemetry telemetryAsRequest,
+            DependencyTelemetry telemetryAsDependency,
+            ExceptionTelemetry telemetryAsException)
```

Comment:
This is a performance optimization. This code is executed for every incoming request, so I want to avoid doing dynamic type conversions more than once.